liota/utilities/si_unit.py

```diff
@@ -0,0 +1,324 @@
+cr_replaces = {
+        re.compile(r"\bmeter\b"): "metre",
+        re.compile(r"\bdegC\b"): "degree Celsius",
+        re.compile(r"Bq\b"): "becquerel",
+        re.compile(r"Gy\b"): "gray",
+        re.compile(r"\*\*\s2\b"): "squared",
+        re.compile(r"\*\*\s3\b"): "cubed",
+        re.compile(r"\*\*\s4\b"): "to the fourth power",
+        re.compile(r"\b1\s\/\s(meter|metre)"): r"reciprocal \1",
+        re.compile(r"\s\*\s"): " "
+    }
+cr_patches = {
+		re.compile(r"metre\ssquared"): "square metre",
+		re.compile(r"metre\scubed"): "cubic metre",
+		re.compile(r"(kelvin|degree\sCelsius)\skilogram"): r"kilogram \1",
+		re.compile(r"(kelvin|degree\sCelsius)\smetre"): r"metre \1",
+		re.compile(r"(kelvin|degree\sCelsius)\smole"): r"mole \1",
+		re.compile(r"metre\snewton"): "newton metre"
+	}
+
+# Find prefix with provided multiplier and return its full name
+def _get_prefix(multiplier):
+    return prefixes[multiplier]
+
+#---------------------------------------------------------------------------
+# Here we defined an error that is raised when we cannot parse user defined
+# unit for whatever reason.
+# Currently, possible reasons include using non-SI unit, having prefix on a
+# squared or cubed unit, and having unsupported power of unit.
+
+class UnsupportedUnitError(ValueError):
+    """
+    Raised when the derived unit is not supported by parser
+    """
+
+    def __init__(self, unit_specs):
+        super(ValueError, self).__init__()
+        self.unit_specs = unit_specs
+
+    def __str__(self):
+        mess = "'{0}' is not a supported unit"
+        return mess.format(self.unit_specs)
+
+#---------------------------------------------------------------------------
+# This method returns name of unit according to SI specs,
+# or return None if provided unit reference is None.
+
+def _get_unit_name(unit):
+    if unit is None:
+        return None
+
+    # Get string of unit from pint
+    ts = str(unit)
+
+    if(ts == "dimensionless"):
+        return None
+    for cr, rp in cr_replaces.items():
+        ts = cr.sub(rp, ts)
+
+    # Detect higher powers and throw exception
+    if re.compile("\*\*\s\-?\d").search(ts) is not None:
+        raise UnsupportedUnitError(unit)
+
+    # Detect numbers and throw exception
+    if re.compile("\d").search(ts) is not None:
+        raise UnsupportedUnitError(unit)    
+
+    # Additional and advanced replacements
+    ts = re.compile(r"\s\/\s").sub(" per ", ts, count=1)
+    ts = re.compile(r"\s\/\s").sub(" ", ts, count=1)
+
+    # Apply name patches
+    for cr, rp in cr_patches.items():
+        ts = cr.sub(rp, ts)
+
+    return ts
+
+#---------------------------------------------------------------------------
+# This is primary method that should be imported in vROps DCC, or any future
+# DCC that has or requires unit support, in order to parse unit parameter
+# passed through create_metric call.
+
+def parse_unit(unit):
+    pf = None
```

Comment:
A lot of names in this CL seem to be non-intuitive (pf, un, ts etc.) and require a lot of effort to understand the meaning. Can you change the variable names?